### PR TITLE
fix: Python Workflow with Incorrect Environ

### PR DESCRIPTION
**Pull request by CoshUS**
*Issue #, if available:*
https://github.com/aws/aws-lambda-builders/issues/224

*Description of changes:*
Patches the LB_LIBRARY_PATH envar and passes the original one to all popens in Python workflow.
This is required as PyInstaller changes LB_LIBRARY_PATH and affects its subprocesses.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ for:
 
     # setup go
     - rmdir c:\go /s /q
-    - "choco install golang"
+    - "choco install golang --version 1.15.7"
```
Comment (CoshUS) [inline .appveyor.yml:63]: Pinning down golang version to 1.15 as 1.16 changes default Windows installation directory which breaks AppVeyor.

**Review by wchengru — DISMISSED** [review]

File: aws_lambda_builders/workflows/python_pip/utils.py [hunk 2]
```diff
@@ -16,6 +16,19 @@ class OSUtils(object):
     def environ(self):
         return os.environ
 
+    def original_environ(self):
+        # https://pyinstaller.readthedocs.io/en/stable/runtime-information.html#ld-library-path-libpath-considerations
+        env = dict(os.environ)
+        lp_key = "LD_LIBRARY_PATH"
+        original_lp = env.get(lp_key + "_ORIG")
```
Comment (sriram-mv) [inline aws_lambda_builders/workflows/python_pip/utils.py:23]: is this just specific to some systems where `LD_LIBRARY_PATH_ORIG` is populated? is there any way we can make this very specific to pyinstaller? does pyinstaller re-introduce any other env vars?

**Review by sriram-mv — CHANGES_REQUESTED** [review]

File: aws_lambda_builders/workflows/python_pip/utils.py (hunk 2 above)
Comment (CoshUS) [reply aws_lambda_builders/workflows/python_pip/utils.py:23]: LD_LIBRARY_PATH_ORIG is only populated if LD_LIBRARY_PATH is set. There are no other env vars that are generated by PyInstaller. I will add a check to run this only in Linux.

**Review by sriram-mv — DISMISSED** [review]
Looks good, could you clean up the commit history?

**Comment by CoshUS** [comment]
Cleaned up merge commits.

**Review by wchengru — APPROVED** [review]

**Review by sriram-mv — APPROVED** [review]

**Pull request merged**